datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxHorizontalScalingService.java

```diff
@@ -64,13 +72,29 @@ public class SdxHorizontalScalingService {
     @Inject
     private CloudbreakFlowService cloudbreakFlowService;
 
+    private static Map<String, Set<InstanceMetaDataV4Response>> getInstanceMap(StackV4Response stack, String target, String gatewayInstanceName) {
+        Map<String, Set<InstanceMetaDataV4Response>> instanceMap = stack.getInstanceGroups().stream()
+                .filter(instance -> instance.getName().equalsIgnoreCase(target) || instance.getName().equalsIgnoreCase(gatewayInstanceName))
+                .collect(Collectors.toMap(instance -> instance.getName().toLowerCase(Locale.ROOT), InstanceGroupV4Response::getMetadata));
+        return instanceMap;
+    }
+
+    private List<InstanceMetaDataV4Response> getGatewayHealthyMetadata(String gatewayInstanceName, Map<String, Set<InstanceMetaDataV4Response>> instanceMap) {
```

Comment:
Forgive me if I misunderstood this - should this method be called `getGatewayUnhealthyMetadata` because it seems it is trying to catch unhealthy data?